CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+# Contributing to dosbox-staging
+
+Thank you for your interest in contributing to dosbox-staging! There are many
+ways to contribute, and we appreciate all of them. This document is a bit long,
+so here's links to the major sections:
+
+- [1 Feature requests and bug reports](#feature-requests-and-bug-reports)
+- [2 Find something to work on](#find-something-to-work-on)
+- [3 Build dosbox-staging](#build-dosbox-staging)
+- [4 Contributing code](#contributing-code)
+   - [4.1 Coding style](#coding-style)
+      - [4.1.1 Language](#language)
+      - [4.1.1 Code formatting]()
+      - [4.1.1 Additional style rules]()
+   - [4.2 Submitting patches / Pull Requests](#submitting-patches-pull-requests)
+      - [4.2.1 Commit messages]()
+      - [4.2.2 Commit messages when you're not author of the patch]()
+- [5 Tools](#tools)
+   - [5.1 Using clang-format](#using-clang-format)
+   - [5.2 Summarize warnings](#summarize-warnings)
```

Comment:
All of this can be form as a sub-section under the Development contribution branch.

Reply:
Comment against just lines 9 and 10:
 - Imagine we have a typical dosbox patch author who knows how to download a .tar.gz, unzip it on Windows, modify the code, runs "diff" on the directories, and then zip's the patch and uploads it to share. 

We need to help these potential contributors bridge the gap to git and github. (now, this isn't for us to invent the wheel; but as we've seen this can be a struggle greater than the code itself. So pointing users to resources and tutorial near the top of this section is probably useful.

Here's decent walk through starting from simply creating an account, all the way through creating your first pull request - https://www.freecodecamp.org/news/how-to-make-your-first-pull-request-on-github-3/    Let me know what you think!

Then, all the nuances and rules about our project (coding style, commit messages, etc can refine their knowledge)

Reply:
All those pieces of information are one simple Google search away… I wanted this document only on things specific to our project.